source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/DateAndTimeUnitTest.cpp

```diff
@@ -11,82 +11,37 @@
 using namespace Microsoft::VisualStudio::CppUnitTestFramework;
 using namespace AdaptiveCards;
 
+// Use this macro to effectively skip a test when run outside of the Pacific timezone
+#define SKIP_TEST_IF_NEEDED() \
+    if (!fShouldRunTests)                       \
+    {                                           \
+        Assert::IsTrue(true, L"Skipping test for non-Pacific timezone"); \
+        return;                                                         \
+    }
+
 namespace AdaptiveCardsSharedModelUnitTest
 {
-    static TIME_ZONE_INFORMATION tzOriginal{};
-    static bool changedTime = false;
-
-    void EnableTimeZoneChange(bool set)
-    {
-        HANDLE hToken;
-        TOKEN_PRIVILEGES tkp;
-
-        OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &hToken);
-        LookupPrivilegeValue(NULL, SE_TIME_ZONE_NAME, &tkp.Privileges[0].Luid);
-        tkp.PrivilegeCount = 1;
-        tkp.Privileges[0].Attributes = set?SE_PRIVILEGE_ENABLED:0;
-        if (!AdjustTokenPrivileges(hToken, FALSE, &tkp, 0, (PTOKEN_PRIVILEGES)NULL, 0))
-        {
-            Logger::WriteMessage("Could not get the permissions to changes the timezone");
-        }
-    }
+    bool fShouldRunTests = false;
 
-    // Time tests assume a Pacific Timezone. Make sure that we're
-    // setting it up appropriately for the run of the tests.
     TEST_MODULE_INITIALIZE(InitializeTimeZone)
     {
-        GetTimeZoneInformation(&tzOriginal);
-        changedTime = false;
-        // Don't do this if we're already on the right timezone
-        if (tzOriginal.Bias != 480)
-        {
-            Logger::WriteMessage("Changing TimeZone to Seattle");
-            TIME_ZONE_INFORMATION tzSeattle = { 0 };
-            tzSeattle.Bias = 480;
-            StringCchCopy(tzSeattle.StandardName, 32, L"Test Standard Zone");
-            tzSeattle.StandardDate.wMonth = 11;
-            tzSeattle.StandardDate.wDayOfWeek = 0;
-            tzSeattle.StandardDate.wDay = 1;
-            tzSeattle.StandardDate.wHour = 2;
-
-            StringCchCopy(tzSeattle.DaylightName, 32, L"Test Daylight Zone");
-            tzSeattle.DaylightDate.wMonth = 3;
-            tzSeattle.DaylightDate.wDayOfWeek = 0;
-            tzSeattle.DaylightDate.wDay = 2;
-            tzSeattle.DaylightDate.wHour = 2;
-            tzSeattle.DaylightBias = -60;
-
-            EnableTimeZoneChange(true);
-            bool timeZoneChanged = SetTimeZoneInformation(&tzSeattle);
-            EnableTimeZoneChange(false);
-            if (!timeZoneChanged)
-            {
-                Assert::Fail(L"Could not change the timezone");
-            }
-            changedTime = true;
-        }
-    }
+        // Make sure we only run our tests in Pacific timezone
+        TIME_ZONE_INFORMATION tz{};
+        GetTimeZoneInformation(&tz);
+        fShouldRunTests = (tz.Bias == 480l);
```

Comment:
>480l [](start = 38, length = 4)

Hmm does the bias here account for both PDT and PST? Do we need to check the StandardBias and DaylightBias fields separately?

Reply:
Also i remembered on the Rome codebase we'd been working with std::chrono for most/all time-related operations --apparently it has timezone helpers? [https://en.cppreference.com/w/cpp/chrono/current_zone](https://en.cppreference.com/w/cpp/chrono/current_zone) ? Maybe? 

---
In reply to: [346021077](https://github.com/microsoft/AdaptiveCards/pull/3611#discussion_r346021077) [](ancestors = 346021077)

Reply:
looks like `std::chrono::current_zone` is a C++20 feature, so it's probably not something we want to take a dependency on just yet :)

---
In reply to: [346023811](https://github.com/microsoft/AdaptiveCards/pull/3611#discussion_r346023811) [](ancestors = 346023811,346021077)

Reply:
also, we don't have to check the `StandardBias` or `DaylightBias` fields as they are intended to be added to `Bias` to get actual offset from UTC (i.e. for Pacific time, `StandardBias` is `0` and `DaylightBias` is `-60`

---
In reply to: [346035923](https://github.com/microsoft/AdaptiveCards/pull/3611#discussion_r346035923) [](ancestors = 346035923,346023811,346021077)